File: Random Code/cpp/Other/ACM ICPC Rocky Mountains/E/pages.cpp
#include <bits/stdc++.h>

using namespace std;

typedef vector<int> vi;
typedef vector<vi> vvi;


void solve(int N) {
  string line; cin >> line;
  istringstream istream(line);
  vector<bool> isPrinted(1005, false);
  bool isFirst = true, readComma = false;
  while (true) {
    int L, R; char c;
    if (readComma && !isFirst) {
      if (!(istream >> c)) {
        break;
      }
    }

    istream >> L >> c;
    readComma = true;

    if (c != '-' && L <= N) {
      isPrinted[L] = true;
      readComma = false;
    } else {
      istream >> R;

      for (int j=L;j<min(N, R)+1;j++) {
        isPrinted[j] = true;
      }
    }
    isFirst = false;
  }

  int ans = 0;
  for (int i=0;i<1000;i++) {
    if (isPrinted[i]) ans++;
  }
  cout << ans << endl;
}

int main() {
  int N;
  while (cin >> N, N) {
    solve(N);
  }
  return 0;
}

File: Random Code/Other/ACM ICPC Rocky Mountains/E/pages.py
while True:
    N = int(input())
    if N == 0:
        break
    isPrinted = [False]*(N+5)

    line = input().split(",")

    for r in line:
        if '-' in r:
            L_R = r.split('-')
            L = int(L_R[0])
            R = int(L_R[1])
            if L > R:
                continue
            else:
                for i in range(L,min(N,R)+1):
                    isPrinted[i] = True
        else:
            L = int(r)
            if L <= N:
                isPrinted[L] = True

    ans = 0
    for i in range(1,N+1):
        if isPrinted[i]:
            ans += 1

    print(ans)
